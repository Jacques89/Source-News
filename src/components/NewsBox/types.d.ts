type NewsBoxProps = {
  article: {
    title: string
    img: string
    content: string
    source_url: string
  }
}