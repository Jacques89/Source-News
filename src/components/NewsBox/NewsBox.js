import React, { Component } from 'react'
import '../../styles/NewsBox.scss'

export default class NewsBox extends Component {
  render () {
    console.log(this.props)
    let article = this.props.article
    let title = article.title
    let description = article.description
    let image = article.image
    let url = article.url

    return (
      <article>
        <h3 className="title">{title}</h3>
        <div className="image" style={{ backgroundImage: `url('${image}')` }}></div>
        <p className="description">
          {description}
        </p>
        <p className="link">
          <a href={url}>Continue reading</a>
        </p>
      </article>
    )
  }
}