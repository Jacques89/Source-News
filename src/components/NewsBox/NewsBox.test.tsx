import NewsBox from './NewsBox'

import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('NewsBox', () => {
  const props = {
    article: {
      title: 'article-title',
      img: 'news.png',
      content: 'News description',
      source_url: 'https://externalsource.com/'
    }
  }

  it('renders the NewsBox component', () => {
    render(<NewsBox {...props} />)
    expect(screen.getByRole('article')).toMatchSnapshot()
  })

  it('should navigate to an external site when link is clicked', () => {
    const { getByText } = render(
      <a href={props.article.source_url}>Continue reading</a>
    )
    const link = getByText('Continue reading')
    userEvent.click(link)
    expect(getByText('Continue reading').getAttribute('href')).toBe(
      props.article.source_url 
    )
  })
})
