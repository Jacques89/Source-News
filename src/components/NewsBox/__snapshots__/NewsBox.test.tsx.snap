// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsBox renders correctly 1`] = `
<article>
  <h3
    class="title"
  >
    article-title
  </h3>
  <div
    class="image"
    style="background-image: url(news.png);"
  />
  <p
    class="description"
  >
    News description
  </p>
  <p
    class="link"
  >
    <a
      href="https://externalsource.com/"
    >
      Continue reading
    </a>
  </p>
</article>
`;
